hooks/gen_docs/gen_docs_components.py

```diff
@@ -62,6 +56,37 @@
 log = logging.getLogger("DocumentationGenerator")
 
 
+def collect_parents_mro(component: type[PipelineComponent]) -> list[str]:
+    """Return a list of a component's parents.
+
+    :param component: Component name in kebab-case
+    :return: List ordered from closest to furthest ancestor,
+        i.e. ``result[0] == component_name``.
+    """
+    bases = []
+    for base in component.mro():
+        if patched_issubclass_of_basemodel(base):
```

Comment:
should this really collect all parents that are `pydantic.BaseModel`? I would assume `PipelineComponent` would be enough for our use case.
could even be a property of `PipelineComponent` imo. what do you think?

Reply:
I meant `BaseDefaultsComponent` instead of `PipelineComponent` of course

Reply:
Agreed, we don't need anything beyond `PipelineComponent`. Here we only collect models that have the `type` property which originates in `BaseDefaulltsComponent`, so it's the only "useless" component that we collect. Still, it would be nice to avoid it, I will think of something.

Reply:
mainly just that having a `<PipelineComponent>.parents` property would be kinda nice

you could put your MRO logic in there

Reply:
The property wouldn't allow for selecting a cutoff for the parents, so instead I made it a `classmethod`